### ISSUE TITLE
Pixel data is not stored as an even byte array when adding pixel data to a DicomFile

### ISSUE THREAD

**Issue by Siyam** [fo-dicom/fo-dicom#1051]
### Expected behavior
When adding pixel data, fo-dicom is able to add appropriate padding to the pixel data to ensure the length is even as required by DICOM specification.

### Actual behavior
Fo-dicom adds the pixel data as is without checking to see if the byte array has an even length as required by DICOM specifications.

### Steps to reproduce the behavior
In order to reproduce this, it generally requires an image to have a byte array of odd length, however it can be verified in the code.  

Here is a sample snippet of now having to check if the byte array length is odd:
```
                DrawingImage image = DrawingImage.FromFile(file.Item2);
                MemoryStream memoryStream = new MemoryStream();
                ImageCodecInfo myImageCodecInfo;
                Encoder myEncoder;
                EncoderParameter myEncoderParameter;
                EncoderParameters myEncoderParameters;
                myImageCodecInfo = ImageCodecInfo.GetImageEncoders().Where(s => s.FormatID == ImageFormat.Jpeg.Guid).FirstOrDefault();
                myEncoder = Encoder.Quality;
                myEncoderParameters = new EncoderParameters(1);
                myEncoderParameter = new EncoderParameter(myEncoder, 25L);
                myEncoderParameters.Param[0] = myEncoderParameter;
                image.Save(memoryStream, myImageCodecInfo, myEncoderParameters);
                memoryStream.Position = 0;
                var byteArray = memoryStream.ToArray();

                byte[] resultBytes;

                if (byteArray.Length % 2 > 0)
                {
                    resultBytes = new byte[byteArray.Length + 1];
                    Array.Copy(byteArray, resultBytes, byteArray.Length);
                }
                else
                {
                    resultBytes = byteArray;
                }
                MemoryByteBuffer memoryByteBuffer = new MemoryByteBuffer(resultBytes);
                memoryStream.Dispose();

                DicomDataset dataset = new DicomDataset(internalTransferSyntax: DicomTransferSyntax.JPEGProcess1)
                {
                    { DicomTag.SOPClassUID, DicomUID.SecondaryCaptureImageStorage },
                    { DicomTag.SOPInstanceUID, DicomUIDGenerator.GenerateDerivedFromUUID() },
                    { DicomTag.StudyInstanceUID, studyData.StudyInstanceUID },
                    { DicomTag.SeriesInstanceUID, DicomUIDGenerator.GenerateDerivedFromUUID() },
                    { DicomTag.BitsAllocated, (ushort)8 },
                    { DicomTag.InstanceNumber, file.Item1.ToString() },
                    { DicomTag.AccessionNumber, studyData.StudyAccessionNumber },
                    { DicomTag.PatientID, studyData.PatientId },
                    { DicomTag.PatientName, $"{studyData.PatientLastName}^{studyData.PatientFirstName}^{studyData.PatientMiddleName}" },
                    { DicomTag.PatientBirthDate, studyData.PatientBirthDate },
                    { DicomTag.PatientSex, studyData.PatientSex },
                    { DicomTag.StudyDate, studyData.StudyDate },
                    { DicomTag.StudyTime, studyData.StudyTime },
                    { DicomTag.Modality, studyData.Modality }
                };


                DicomPixelData dicomPixelData = DicomPixelData.Create(dataset: dataset, newPixelData: true);
                dicomPixelData.BitsStored = 8;
                dicomPixelData.SamplesPerPixel = 3;
                dicomPixelData.HighBit = 7;
                dicomPixelData.PixelRepresentation = 0;
                dicomPixelData.PlanarConfiguration = 0;
                dicomPixelData.Height = (ushort)image.Height;
                dicomPixelData.Width = (ushort)image.Width;
                dicomPixelData.PhotometricInterpretation = PhotometricInterpretation.YbrFull422;
                image.Dispose();
                dicomPixelData.AddFrame(memoryByteBuffer);
                DicomFile dicomFile = new DicomFile(dataset);
                dicomFile.FileMetaInfo.TransferSyntax = DicomTransferSyntax.JPEGProcess1;
                dicomFiles.Add(dicomFile);
```

### fo-dicom version and OS/platform
4.05, win 10, .net core3.1

**Comment by gofal** [comment]
do you think this could be connected to #1019  and #1012 ?

**Comment by Siyam** [comment]
@gofal Yes same issue it would appear.  I think it wouldn't be unreasonable for fo-dicom to make this check, or at least produce a warning in the log output.

**Comment by gofal** [comment]
this PR #1410 could be a fix for that issue

**Comment by gofal** [comment]
This issue should be fixed with PR #1474 , please verify.
If with the latest version of fo-dicom there is still an issue, please reopen this issue and provide actual information.
